[Mellanox] Dynamically fetch sensors.conf from hw-management at runtime

YairRaviv opened the pull request:
**Why I did it**
To eliminate duplication and ensure platforms always use up-to-date sensors.conf from hw-management instead of maintaining static sensors.conf files in SONiC.

**How I did it**

- Removed static sensors.conf files from Mellanox SN5600, SN5610n, SN5640 platform directories 
- Added logic in docker_image_ctl.j2 to:
     1. Create a symlink to lm_sensors_config if no sensors.conf exists.
     2. Append SONiC-specific overrides from sonic_specific_sensors.conf if present.

**How to verify it**

- Check that the symlink is created.
- Run 'sensors' command and validate the output.

YairRaviv committed:
[Mellanox] Dynamically fetch sensors.conf from hw-management at runtime

mssonicbld committed:
[submodule] Update submodule sonic-platform-daemons to the latest HEAD automatically (#23259)

#### Why I did it
src/sonic-platform-daemons
```
* fa41b9c - (HEAD -> master, origin/master, origin/HEAD) [Smartswitch][pcied] Fix pcied handling for smartswitch during DPU detach #5 (#633) (9 hours ago) [Gagan Punathil Ellath]
```
#### How I did it
#### How to verify it
#### Description for the changelog

mssonicbld committed:
[submodule] Update submodule sonic-dash-ha to the latest HEAD automatically (#23211)

#### Why I did it
src/sonic-dash-ha
```
* ff1b02a - (HEAD -> master, origin/master, origin/HEAD) Implement HA-Scope actor (#73) (20 hours ago) [yue-fred-gao]
* 832e67a - Implement HA-SET actor (#70) (4 days ago) [yue-fred-gao]
* cadfa46 - Implement vdpu actor (#67) (5 days ago) [yue-fred-gao]
```
#### How I did it
#### How to verify it
#### Description for the changelog

mssonicbld committed:
[submodule] Update submodule sonic-swss-common to the latest HEAD automatically (#23260)

#### Why I did it
src/sonic-swss-common
```
* 863f4da - (HEAD -> master, origin/master, origin/HEAD) add DASH port map table names (#1030) (20 minutes ago) [Lawrence Lee]
* 8062a06 - [azp] Only grab libyang v1 libs for AZP jobs (#1031) (23 hours ago) [Lawrence Lee]
```
#### How I did it
#### How to verify it
#### Description for the changelog

michaelc-nexthop committed:
Fix typo in pddf_current_sensor constructor (#22997)

Line #27 self.display_name = self.sensor_data should be self.display_name = self.sensor_obj

ram25794 committed:
[Nvidia-Bluefield] Fix DPU config error (#22798)

- Why I did it
Clearing Ethernet0 IP for Nvidia Bluefield DPU to avoid re-configuring IP during config reload.

- How I did it
Flushed the IP for Ethernet0 Interface during de-initialization of syncd.

- How to verify it
Tested the code changes on the DPU, and the configuration failure error (sonic ERR swss#intfmgrd: :- setIntfIp: Command '/sbin/ip address "add") no longer occurs.



Signed-off-by: ram25794 <ssingamala@nvidia.com>

lotus-nexthop committed:
[PDDF] Fix Failed to register i2c client optoe2 (#22763)

Gfrom2016 committed:
Upgrade xgs SAI version to 13.2.1.1 (#23263)

Upgrade XGS SAI to 13.2.1.1

noaOrMlnx commented on files/build_templates/docker_image_ctl.j2:
```diff
@@ -685,6 +703,7 @@ start() {
         -e SX_API_SOCKET_FILE=/var/run/sx_sdk/sx_api.sock \
 {%- elif docker_container_name == "pmon" %}
         -v /var/run/hw-management:/var/run/hw-management:rw \
+        -v /etc/hw-management-sensors:/etc/hw-management-sensors:rw \
```
please check if we can remove the mount: 
1. copy the file instead of symlink
2. you can keep with the symlink, but I think we can have read-only mount, and not read-write.

noaOrMlnx commented on files/build_templates/docker_image_ctl.j2:
```diff
@@ -659,6 +659,24 @@ start() {
+        if [ -e "$SONIC_SPECIFIC_SENSORS_CONF" ]; then
+            if ! sudo sh -c "cat '$SONIC_SPECIFIC_SENSORS_CONF' >> '$SONIC_SENSORS_CONF_LINK'"; then
+                echo "Error: Failed to append SONiC-specific sensors.conf"
+                return 1
```
please reconsider the return 1 - you may keep the sensors file without platform specific code

nimicho committed:
yml file changes for Nokia D4 platform (#23079)

1)   MDIO clause45.  DNA DEVICE_CONFIG:VARIANT required only for BCM template profiles/CS00012403483
2)   Add egress_pkt_pri_remark_mode to introduce split mode so as to remark only egress packet priority

Signed-off-by: michos <nikolaos.michos@nokia.com>

mssonicbld committed:
[submodule] Update submodule sonic-mgmt-common to the latest HEAD automatically (#23270)

#### Why I did it
src/sonic-mgmt-common
```
* 0ca4d8d - (HEAD -> master, origin/master, origin/HEAD) [ci] Upgrade agent pool from ubuntu 20.04 to latest. (#173) (12 hours ago) [Liu Shilong]
```
#### How I did it
#### How to verify it
#### Description for the changelog

mssonicbld committed:
[submodule] Update submodule sonic-platform-daemons to the latest HEAD automatically (#23273)

#### Why I did it
src/sonic-platform-daemons
```
* fd863c1 - (HEAD -> master, origin/master, origin/HEAD) Add high power class support for non-CMIS modules in xcvrd (#574) (5 hours ago) [longhuan-cisco]
```
#### How I did it
#### How to verify it
#### Description for the changelog

rick-arista committed:
Add additional config for Arista-7060X6-64PE-B-C512S2, Arista-7060X6-64PE-B-C448O16 (#23126)

Add additional config for Arista-7060X6-64PE-B-C512S2, Arista-7060X6-64PE-B-C448O16

vivekrnv committed:
[nvidia-bluefield] Update platform drivers to use DKMS infrastructure (#23163)

- Why I did it
To always build drivers from sources and don't use pre-built drivers

- How I did it
Saves DKMS for all modules and use those dkms packages to build the modules

gpio-mlxbf3-dkms_1.0-0_all.deb
mlxbf-gige-dkms_1.0-0_all.deb
mlxbf-pka-dkms_2.0-1_all.deb
mlxbf-ptm-dkms_1.0-0_all.deb
mlxbf-tmfifo-dkms_1.0-0_all.deb
mlx-bootctl-dkms_1.3-0_all.deb
pinctrl-mlxbf3-dkms_1.0-0_all.deb
pwr-mlxbf-dkms_1.0-0_all.deb
sdhci-of-dwcmshc-dkms_1.0-0_all.deb
mlnx-ofed-kernel-dkms_25.04.OFED.25.04.0.6.1.1-1_all.deb
Update the version of bluefield-platform-modules package to 2.0 bluefield-platform-modules_2.0_arm64.deb

- How to verify it
Make sure compilation, deployment and basic tests are successful and verify all the drivers are present in bluefield-platform-modules_2.0_arm64.deb

Signed-off-by: Vivek Reddy <vkarri@nvidia.com>

gpunathilell committed:
[Smartswitch][Pmon] Added changes relevant to pcie attach and detach and sensord to support both light and dark modes (#22778)

- Why I did it
Add ignore configuration for sensors so that sensord is restarted after removing relevant sensors on DPU power off
Add skip_pre_post to module implementation of DPU, this is needed because we are handling the pre shutdown and post startup as part of the chassisd and reboot implementation separately, so it is skipped when called from chassisd, but the dpuctl command line will still work as we are calling by not skipping the pre and post calls

- How I did it
Changes relevant to smartswitch pcie detach and attach, and sensord relevant changes

- How to verify it
Tested with image on 202505 base branch manually, updated the test cased and observed that it is passing, and the mlnx-platform-api wheel file is built:

tests/test_module.py::TestModule::test_dpu_module PASSED                 [ 52%]
[ building ] [ target/python-wheels/bookworm/mlnx_platform_api-1.0-py3-none-any.whl ] 
[ finished ] [ target/python-wheels/bookworm/mlnx_platform_api-1.0-py3-none-any.whl ]

Junchao-Mellanox commented on device/mellanox/x86_64-nvidia_sn5600-r0/sonic_specific_sensors.conf:
```diff
@@ -0,0 +1 @@
+# Hardware revision default
```
Make this a soft link?

Junchao-Mellanox commented on files/build_templates/docker_image_ctl.j2:
I think you also need to mount /etc/hw-management-sensors to pmon container

Junchao-Mellanox commented on files/build_templates/docker_image_ctl.j2:
```diff
@@ -659,6 +659,23 @@ start() {
+    HW_MGMT_SENSORS_CONF="/var/run/hw-management/config/lm_sensors_config"
+    SONIC_SENSORS_CONF="/usr/share/sonic/device/$PLATFORM/sensors.conf"
+    SONIC_SPECIFIC_SENSORS_CONF="/usr/share/sonic/device/$PLATFORM/sonic_specific_sensors.conf"
+    if [ -e "$HW_MGMT_SENSORS_CONF" ] && [ ! -e "$SONIC_SENSORS_CONF" ]; then
```
If HW_MGMT_SENSORS_CONF does not exist, we need print an error. BTW, hw-management need time to create files under /var/run/hw-management folder. Is it possible that /var/run/hw-management is not yet ready while pmon.sh is trying to access it? If yes, we have a timing issue here. 

Another option is to move the logic to file dockers/docker-platform-monitor/lm-sensors.sh

yuazhe commented on files/build_templates/docker_image_ctl.j2:
```diff
@@ -659,6 +659,23 @@ start() {
+            echo "Error: Failed to copy sensors.conf file from hw-management, 'sensors' command will print raw sensors data"
+            break
+        fi
+        if [ -e "$SONIC_SPECIFIC_SENSORS_CONF" ]; then
```
I don't think you need this specific sensors conf anymore, it should be possible to determine the current platform hardware revision in the run-time, and you can append it to the final sensors conf in turn.

mssonicbld committed:
[submodule] Update submodule sonic-linux-kernel to the latest HEAD automatically (#23285)

#### Why I did it
src/sonic-linux-kernel
```
* d733205 - (HEAD -> master, origin/master, origin/HEAD) Merge pull request #490 from liushilongbuaa/fix-url (23 hours ago) [Jianquan Ye]
* adb4f0e - [build] Use package.trafficmanager instead of sonicstorage.blob (2 days ago) [Liu Shilong]
```
#### How I did it
#### How to verify it
#### Description for the changelog

YairRaviv committed:
Moved sensors.conf fetching logic to pmon docker_init to ensure the existence of hw-mgmt sensors.conf
